src/internal/transactionenv/env.go

```diff
@@ -82,11 +116,17 @@ func New() *TransactionEnv {
 
 // Initialize stores the references to APIServer instances in the TransactionEnv
 func (tnxEnv *TransactionEnv) Initialize(
-	serviceEnv serviceenv.ServiceEnv,
+	db *sqlx.DB,
+	getAuth func() AuthBackend,
+	getPFS func() PFSBackend,
+	getPPS func() PPSBackend,
 	txnServer TransactionServer,
 ) {
-	tnxEnv.serviceEnv = serviceEnv
+	tnxEnv.db = db
 	tnxEnv.txnServer = txnServer
+	tnxEnv.getPFS = getPFS
```

Comment:
Do these things need to be functions?  I feel like they always return the same thing.

Reply:
Once I get the build green, I'll try that. 

It would be nice, but my prior for this codebase is that some extremely late setup happens in some cases and it needs to be a closure.

Reply:
I believe that.  I don't really know why we do the lazy setup necessarily, sure connect to k8s and etcd in parallel, but then just build the clients:

```
pps := &PPS{}
pfs := &PFS{}
pps.PFS = pfs
pfs.PPS = pps
```

Doesn't have to be done in this PR obviously.

Reply:
😦 It's not looking good for the sans-closures version.  Maybe those are flakes, we shall see.